goembed.go

```diff
@@ -66,14 +111,34 @@ import (
 )
 
 func init() {
-	r, err := gzip.NewReader(bytes.NewReader({{.}}_gzip))
+	var (
+		r *gzip.Reader
+		err error
+	)
+
+{{ if gt (len .Args) 0 }}
+{{ range $idx, $var := .Args }}
+{{ $n := printf "%s_%d" $.VarName $idx }}
+	r, err = gzip.NewReader(bytes.NewReader({{ $n }}_gzip))
+	if err != nil {
+		panic(err)
+	}
+	{{ $n }}, err = ioutil.ReadAll(r)
+	r.Close()
```

Comment:
Closing the `gzip.Reader` directly so that the resource overhead doesn’t pile up as the number of files grows.